src/Compilers/CSharp/Test/Emit3/Semantics/UnsafeEvolutionTests.cs

```diff
@@ -648,4 +766,938 @@ public void Pointer_ElementAccess_UnsafeContext()
+            //     System.Span<int> a = stackalloc int[5];
+            Diagnostic(ErrorCode.ERR_UnsafeUninitializedStackAlloc, "stackalloc int[5]").WithLocation(8, 26),
+            // (11,26): error CS0847: An array initializer of length '3' is expected
+            //     System.Span<int> e = stackalloc int[3] { 1, 2 };
+            Diagnostic(ErrorCode.ERR_ArrayInitializerIncorrectLength, "stackalloc int[3] { 1, 2 }").WithArguments("3").WithLocation(11, 26),
+        };
+
+        CreateCompilationWithSpan(source, options: TestOptions.UnsafeReleaseExe.WithUpdatedMemorySafetyRules())
+            .VerifyDiagnostics(expectedDiagnostics);
+
+        CreateCompilationWithSpan(source,
+            parseOptions: TestOptions.RegularNext,
+            options: TestOptions.UnsafeReleaseExe.WithUpdatedMemorySafetyRules())
+            .VerifyDiagnostics(expectedDiagnostics);
+
+        CreateCompilationWithSpan(source,
+            parseOptions: TestOptions.Regular14,
+            options: TestOptions.UnsafeReleaseExe.WithUpdatedMemorySafetyRules())
+            .VerifyDiagnostics(
+            // (1,1): error CS8652: The feature 'updated memory safety rules' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+            // int* x = stackalloc int[3];
+            Diagnostic(ErrorCode.ERR_FeatureInPreview, "int*").WithArguments("updated memory safety rules").WithLocation(1, 1),
+            // (1,10): error CS8652: The feature 'updated memory safety rules' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+            // int* x = stackalloc int[3];
+            Diagnostic(ErrorCode.ERR_FeatureInPreview, "stackalloc int[3]").WithArguments("updated memory safety rules").WithLocation(1, 10),
+            // (8,26): error CS9501: stackalloc expression without an initializer inside SkipLocalsInit may only be used in an unsafe context
+            //     System.Span<int> a = stackalloc int[5];
+            Diagnostic(ErrorCode.ERR_UnsafeUninitializedStackAlloc, "stackalloc int[5]").WithLocation(8, 26),
+            // (11,26): error CS0847: An array initializer of length '3' is expected
+            //     System.Span<int> e = stackalloc int[3] { 1, 2 };
+            Diagnostic(ErrorCode.ERR_ArrayInitializerIncorrectLength, "stackalloc int[3] { 1, 2 }").WithArguments("3").WithLocation(11, 26));
+    }
+
+    [Fact]
+    public void StackAlloc_UnsafeContext()
+    {
+        var source = $$"""
+            int* x = stackalloc int[3];
+            System.Span<int> y = stackalloc int[5];
+            M();
+
+            [System.Runtime.CompilerServices.SkipLocalsInit]
+            void M()
+            {
+                unsafe { System.Span<int> a = stackalloc int[5]; }
+                System.Span<int> b = stackalloc int[] { 1 };
```

Comment:
Is it still useful to keep `b` and `d` in the "UnsafeContext" test, since they are not inside an `unsafe` context?